rktlet/cli/init.go

```diff
@@ -35,21 +36,54 @@ func NewSystemd(systemdRunPath string, execer utilexec.Interface) Init {
 	return &systemd{systemdRunPath, execer}
 }
 
+// cgroupParentToSliceName converts a cgroup path such as:
+//   /kubepods.slice/kubepods-besteffort.slice/kubepods-besteffort-pod5c5979ec-9871-11e7-b58f-c85b763781a4.slice
+// into a systemd slice name such as:
+//   hubepods-besteffort-pod5c5979ec-9871-11e7-b58f-c85b763781a4.slice
+//
+// The systemd slice name must observe the following rules:
+// - the name does not start with a "/" (otherwise that's interpreted as a mount unit)
+// - the dashes ("-") are representing subdirectories
+// - the name finishes with ".slice"
+//
+// Unfortunately, the Kubelet might not give a cgroupParent that is
+// convertible to the systemd slice naming. For example:
+//   /kubepods/besteffort/pod5c5979ec-9871-11e7-b58f-c85b763781a4
+// In that case, do the best we can and issue a warning.
+func cgroupParentToSliceName(cgroupParent string) string {
+	// Example for podDir: "pod5c5979ec-9871-11e7-b58f-c85b763781a4"
+	podDir := path.Base(cgroupParent)
+
+	slice := ""
+	if strings.HasPrefix(podDir, "pod") {
```

Comment:
If the cgroupParent is supported by systemd, it will start with `kubepods-...`, right? That's why checking for `"pod"` works here.

Reply:
I changed the logic here